### PR TITLE
chore: bump pyarrow

### PR DESCRIPTION
**Pull request by villebro**
### SUMMARY
With the 1.0 release of PyArrow featuring semver API versioning and stabilized columnar format, it makes sense to bump PyArrow to the latest version.

Changelog:
- [Arrow 1.0.0](https://arrow.apache.org/blog/2020/07/24/1.0.0-release/)

Edit: There appears to be a potential regression in the Pandas 1.1 branch causing trouble with `MultiIndex`es. Therefore Pandas is only bumped to 1.0.5 until the `MultiIndex` problem has been properly addressed.

### TEST PLAN
CI

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/base.py [hunk 1]
```diff
@@ -441,17 +441,6 @@ def set_or_update_query_limit(cls, sql: str, limit: int) -> str:
...
         return parsed_query.set_or_update_query_limit(limit)
 
-    @staticmethod
-    def excel_to_df(**kwargs: Any) -> pd.DataFrame:
-        """ Read excel into Pandas DataFrame
-           :param kwargs: params to be passed to DataFrame.read_excel
-           :return: Pandas DataFrame containing data from excel
-        """
-        kwargs["encoding"] = "utf-8"
-        kwargs["iterator"] = True
```
Comment (villebro) [inline superset/db_engine_specs/base.py:451]: `encoding` and `iterator` are not supported on `read_excel`: 
https://pandas.pydata.org/docs/reference/api/pandas.read_excel.html#pandas.read_excel

File: superset/db_engine_specs/base.py [hunk 2]
```diff
@@ -521,7 +510,7 @@ def create_table_from_excel(  # pylint: disable=too-many-arguments
         Create table from contents of a excel. Note: this method does not create
         metadata for the table.
         """
-        df = cls.excel_to_df(io=filename, **excel_to_df_kwargs,)
+        df = pd.read_excel(io=filename, **excel_to_df_kwargs)
```
Comment (villebro) [inline superset/db_engine_specs/base.py:513]: Removing method `excel_to_df` as it no longer does anything beyond calling `pd.read_excel()`.

**Review by mistercrunch — APPROVED** [review]

**Review by bkyryliuk — APPROVED** [review]

**Comment by john-bodley** [comment]
@villebro something we're trying at Airbnb for a number of our Python services is automatically updating all Python dependencies on a regular occurrence. I was wondering whether there is merit in doing something similar for Apache Superset especially as _all_ Python dependencies should be defined in either there requirements or `pre-commit-config.yaml`; both of which can be updated via the CLI.

**Comment by villebro** [comment]
@john-bodley what mechanism are you using for automatic updating? Dependabot seems to do a good job for JS and apparently [supports Python](https://dependabot.com/python/), perhaps that could be an option?

**Comment by john-bodley** [comment]
@villebro we have an internal system for running the updates and a monthly cron. It seems like `pip-compile-multi` uses a GitHub workflow which we could replicate ([example](https://github.com/peterdemin/pip-compile-multi/blob/master/.github/workflows/pipcompilemulti.yml)) and run at a weekly cadence or similar. The commands we would need to run are:

```
pip-compile-multi
pre-commit autoupdate —freeze
```

**Pull request merged**